### PR TITLE
Add missing comments to localized strings, reorder buttons, mark dest…

### PR DESCRIPTION
**Pull request by tyiu**
…ructive buttons, prevent blocking or reporting yourself

### CODE REVIEW

File: damus/ContentView.swift [hunk 1]
```diff
@@ -472,20 +477,18 @@ struct ContentView: View {
...
-            Text("No block list found, create a new one? This will overwrite any previous block lists.")
+            Text("No block list found, create a new one? This will overwrite any previous block lists.", comment: "Alert message prompt that asks if the user wants to create a new block list, overwriting previous block lists.")
         })
-        .alert("Block User", isPresented: $confirm_block, actions: {
-            Button("Block") {
+        .alert(NSLocalizedString("Block User", comment: "Title of alert for blocking a user."), isPresented: $confirm_block, actions: {
+            Button(NSLocalizedString("Cancel", comment: "Alert button to cancel out of alert for blocking a user."), role: .cancel) {
+                confirm_block = false
+            }
+            Button(NSLocalizedString("Block", comment: "Alert button to block a user."), role: .destructive) {
```
Comment (tyiu) [inline damus/ContentView.swift:487]: Swapped order of buttons and marked destructive button.
Simulator Screen Shot - iPhone 14 Pro - 2023-01-26 at 23 07 26

File: damus/Views/Events/EventMenu.swift [hunk 2]
```diff
@@ -38,14 +38,14 @@ struct EventMenuContext: View {
             Label(NSLocalizedString("Copy Note JSON", comment: "Context menu option for copying the JSON text from the note."), systemImage: "square.on.square")
         }
         
-        Button {
+        Button(role: .destructive) {
```
Comment (tyiu) [inline damus/Views/Events/EventMenu.swift:41]: Simulator Screen Shot - iPhone 14 Pro - 2023-01-26 at 23 01 21

File: damus/Views/ProfileView.swift [hunk 3]
```diff
@@ -374,17 +374,17 @@ struct ProfileView: View {
...
         }
-        .confirmationDialog("Actions", isPresented: $action_sheet_presented) {
-            Button("Share") {
+        .confirmationDialog(NSLocalizedString("Actions", comment: "Title for confirmation dialog to either share, report, or block a profile."), isPresented: $action_sheet_presented) {
+            Button(NSLocalizedString("Share", comment: "Button to share the link to a profile.")) {
                 show_share_sheet = true
             }
             
-            Button("Report") {
+            Button(NSLocalizedString("Report", comment: "Button to report a profile."), role: .destructive) {
```
Comment (tyiu) [inline damus/Views/ProfileView.swift:382]: Simulator Screen Shot - iPhone 14 Pro - 2023-01-26 at 23 01 45

**Comment by tyiu** [comment]
Tacked on an additional commit to prevent blocking or reporting yourself.
Simulator Screen Shot - iPhone 14 Pro - 2023-01-26 at 23 35 37
hubusercontent.com/963907/215010363-b1c2d926-0664-4415-a6c9-2c17ec1cf6c7.png)

**Comment by jb55** [comment]
thanks!

**Pull request merged**